### ISSUE TITLE
argument of type 'AttributeError' is not iterable

### ISSUE THREAD

**Issue by smk4664** [nautobot/nautobot#4341]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.0-rc (1 and 2)
* Python version: 3.10
* Database platform, version: MySQL...
* Middleware(s):

### Steps to Reproduce
1. Create a Model with no detail view
2. Set default_return_url to the List View
3. Create a record for the model using the Add form.

### Expected Behavior

Object is created and I am redirected back to the list view.

### Observed Behavior

TypeError: argument of type 'AttributeError' is not iterable raised by Django/shortcuts.py
/opt/nautobot/lib/python3.10/site-packages/django/shortcuts.py, line 136, in resolve_url

I expect this is due to the function `get_absolute_url` in BaseModel returning AttributeError instead of raising the error.

**Comment by HanlinMiao** [comment]
Yeah, the problematic code is in `nautobot.core.models.__init__.BaseModel`.

```
def get_absolute_url(self, api=False):
        """
        Return the canonical URL for this object in either the UI or the REST API.
        """

        # Iterate the pk-like fields and try to get a URL, or return None.
        fields = ["pk"]
        actions = ["retrieve", "detail", ""]  # TODO: Eventually all retrieve

        for field in fields:
            if not hasattr(self, field):
                continue

            for action in actions:
                route = get_route_for_model(self, action, api=api)

                try:
                    return reverse(route, kwargs={field: getattr(self, field)})
                except NoReverseMatch:
                    continue

        return AttributeError(f"Cannot find a URL for {self} ({self._meta.app_label}.{self._meta.model_name})")
```

**Comment by bryanculver** [comment]
@smk4664 

What is your use case for a "Model with no detail view"?

And if `get_absolute_url` raised an error (which I agree with your assertion), would that resolve your issue?

**Comment by smk4664** [comment]
I believe that raising the error will resolve the issue as the django/shortcuts.py:resolve_url will not catch the error, nor try to process the return value of `get_absolute_url` but instead the get_return_url function would then catch the AttributeError and return the `default_return_url`.

As to my use case: These are models that inherit the BaseModels class in ChatOps that have never had a detail view for them.  A detail(retrieve) view can be created for them, but not currently scoped for the 2.0 migration.

**Comment by smk4664** [comment]
I do have another workaround, just wanted to bring this up as I feel it was unintentional bug, not an intentional breaking change.
